endaq/calc/utils.py

```diff
@@ -56,3 +58,48 @@ def logfreqs(
         np.log2(1 / dt) - 1,
         1 / bins_per_octave,
     )
+
+
+def to_dB(data: np.ndarray, reference: float, squared=False):
+    """
+    Scale data into units of decibels.
+
+    Decibels are a log-scaled ratio of some value against a reference;
+    typically this is expressed as follows:
+
+    ..math:: dB = 10 \\log10\\left( \\frac{x}{x_{\\text{ref}}} \\right)
+
+    By convention, "decibel" units tend to operate on units of *power*. For
+    units that are proportional to power *when squared* (e.g., volts, amps,
+    pressure, etc.), their "decibel" representation is typically doubled (i.e.,
+    :math:`dB = 10 \\log20(...)`). Users can specify which scaling to use
+    with the `squared` parameter.
+
+    ..note:: Decibels can**NOT** be calculated from negative values.
```

Comment:
small typo, I think
```suggestion
    ..note:: Decibels can **NOT** be calculated from negative values.
```

Reply:
thanks for the thorough review. but actually that was intended; I wanted to bold just the last half of "cannot". Do you think I should change it?